Legacy/bonej/src/main/java/org/bonej/plugins/ParticleAnalysis.java

```diff
@@ -692,10 +745,38 @@ private static double getSurfaceArea(final List<Point3f> points) {
 			Object[] ellipsoid = null;
 			try {
 				ellipsoid = FitEllipsoid.yuryPetrov(coOrdinates);
-			} catch (final RuntimeException re) {
-
+			} catch (final IllegalArgumentException re) {
+				IJ.log("Could not fit ellipsoid to particle: "+re.getMessage());
+			} catch (final Exception e) {
```

Comment:
Should only catch `IllegalArgumentException` because that's the only thing we know `yuryPetrov` can throw. If it suddenly starts throwing something else, we're in trouble and should be alerted by seeing the exception come up.

Reply:
There's also a RuntimeException that gets thrown by JAMA if the Matrix is singular, in `LUDecomposition.solve()`:
```java
	/**
	 * Solve A*X = B
	 *
	 * @param B
	 *            A Matrix with as many rows as A and any number of columns.
	 * @return X so that L*U*X = B(piv,:)
	 * @exception IllegalArgumentException
	 *                Matrix row dimensions must agree.
	 * @exception RuntimeException
	 *                Matrix is singular.
	 */

	public Matrix solve(final Matrix B) {
		if (B.getRowDimension() != m) {
			throw new IllegalArgumentException("Matrix row dimensions must agree.");
		}
		if (!this.isNonsingular()) {
			throw new RuntimeException("Matrix is singular.");
		}
```
This might better be caught and handled by `yuryPetrov()`, but still it will get passed up the chain to client code that needs to know that an ellipsoid could not be fitted to the points it submitted to `yuryPetrov()`.

Reply:
In that case you can write
```
case (final IllegalArgumentException | RuntimeException e) {
    IJ.log("Could not fit ellipsoid to particle: "+e.getMessage());
}
```